src/main/java/uk/ac/ebi/ampt2d/accession/serial/block/MonotonicRange.java

```diff
@@ -0,0 +1,171 @@
+        if (value == 0) {
+            return Long.compare(end, monotonicRange.end);
+        }
+        return value;
+    }
+
+    /**
+     * Returns a array of monotonically increasing ids from start to end inclusively. The array is generated and
+     * returned on demand to make the life span of the array as short as possible in order to have a smaller memory
+     * footprint
+     *
+     * @return
+     */
+    public long[] getIds() {
+        int size = getTotalOfValues();
+        long[] ids = new long[size];
+        long tempId = start;
+        for (int i = 0; i < size; i++) {
+            ids[i] = tempId;
+            tempId++;
+        }
+        return ids;
+    }
+
+    public long getStart() {
+        return start;
+    }
+
+    public long getEnd() {
+        return end;
+    }
+
+    public int getTotalOfValues() {
+        return Math.toIntExact((end - start + 1));
+    }
+
+    public Pair<MonotonicRange, MonotonicRange> split(int numberOfElements) {
+        return Pair.of(new MonotonicRange(start, start + numberOfElements - 1),
+                new MonotonicRange(start + numberOfElements, end));
+    }
+
+    public static List<MonotonicRange> convertToMonotonicRanges(long... accessions) {
+        List<MonotonicRange> ranges = new ArrayList<>();
+
+        if (accessions.length == 0) {
+            return ranges;
+        }
+
+        Arrays.sort(accessions);
+        long minMonotonicValue = accessions[0];
+        long maxMonotonicValue = accessions[0];
+        for (int i = 1; i < accessions.length; i++) {
+            if (maxMonotonicValue + 1 == accessions[i]) {
+                maxMonotonicValue = accessions[i];
+            } else {
+                ranges.add(new MonotonicRange(minMonotonicValue, maxMonotonicValue));
+                minMonotonicValue = accessions[i];
+                maxMonotonicValue = accessions[i];
+            }
+        }
+        ranges.add(new MonotonicRange(minMonotonicValue, maxMonotonicValue));
+        return ranges;
+    }
+
+    public List<MonotonicRange> excludeIntersections(List<MonotonicRange> ranges) {
+        //Sorted ensures that start values are sorted monotonically
+        List<MonotonicRange> intersectingRanges = ranges.stream().filter(this::intersects).sorted()
+                .collect(Collectors.toList());
+        long i = start;
+        List<MonotonicRange> result = new ArrayList<>();
+        if (intersectingRanges.isEmpty()) {
+            result.add(new MonotonicRange(start, end));
+            return result;
+        }
+        for (MonotonicRange range : intersectingRanges) {
+            // If the rest of the elements will be excluded stop now
+            if (i > range.end) {
+                break;
+            }
+            if (i < range.start) {
+                result.add(new MonotonicRange(i, range.start - 1));
+            }
+            if (i > range.start) {
+                i = Math.max(i, range.end + 1);
+                continue;
```

Comment:
I'm not a fan of continue, and the second if works but it could be an "else if" and would look less awkward (in my opinion). I got this to avoid the use of 'continue', but, to be honest, I don't like it either:
 ```
            if (i > range.start) {
                i = Math.max(i, range.end + 1);
            } else {
                if (i < range.start) {
                    result.add(new MonotonicRange(i, range.start - 1));
                }
                i = range.end + 1;
            }
```
If somebody else don't like continue and get a better solution, please comment, if not, forget it!